hls/decode-util_test.go

```diff
@@ -2,37 +2,101 @@ package hls
 
 import (
 	"bufio"
+	"fmt"
 	"os"
 	"reflect"
 	"testing"
 	"time"
 )
 
-func TestV3Compat(t *testing.T) {
+func TestProtocolCompat(t *testing.T) {
 	tests := []struct {
 		expectErr bool
+		mediaType string
 		file      string
 	}{
 		{
 			expectErr: true,
+			mediaType: "media",
 			file:      "fixture-v3-fail.m3u8",
 		},
 		{
 			expectErr: false,
+			mediaType: "media",
 			file:      "fixture-v3.m3u8",
 		},
+		{
+			expectErr: true,
+			mediaType: "media",
+			file:      "fixture-v4-byterange-fail.m3u8",
+		},
+		{
+			expectErr: true,
+			mediaType: "media",
+			file:      "fixture-v4-iframes-fail.m3u8",
+		},
+		{
+			expectErr: false,
+			mediaType: "media",
+			file:      "fixture-v4.m3u8",
+		},
+		{
+			expectErr: true,
+			mediaType: "media",
+			file:      "fixture-v5-keyformat-fail.m3u8",
+		},
+		{
+			expectErr: true,
+			mediaType: "media",
+			file:      "fixture-v5-map-fail.m3u8",
+		},
+		{
+			expectErr: false,
+			mediaType: "media",
+			file:      "fixture-v5.m3u8",
+		},
+		{
+			expectErr: true,
+			mediaType: "media",
+			file:      "fixture-v6-map-no-iframes-fail.m3u8",
+		},
+		{
+			expectErr: false,
+			mediaType: "media",
+			file:      "fixture-v6.m3u8",
+		},
+		{
+			expectErr: true,
+			mediaType: "master",
+			file:      "fixture-v7-media-service-fail.m3u8",
+		},
+		{
+			expectErr: false,
+			mediaType: "master",
+			file:      "fixture-v7.m3u8",
+		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.file, func(t *testing.T) {
-			f, err := os.Open("./testdata/" + tt.file)
+			f, err := os.Open(fmt.Sprintf("./testdata/%s/%s", tt.mediaType, tt.file))
```

Comment:
Could probably use https://golang.org/pkg/path/filepath/#Join here if you liked.